src/Adapter/MSTest.TestAdapter/Execution/UnitTestRunner.cs

```diff
@@ -180,7 +180,7 @@ internal UnitTestResult[] RunSingleTest(TestMethod testMethod, IDictionary<strin
                         // Run the test method
                         testContextForTestExecution.SetOutcome(testContextForClassInit.Context.CurrentTestOutcome);
                         var testMethodRunner = new TestMethodRunner(testMethodInfo, testMethod, testContextForTestExecution);
-                        result = testMethodRunner.Execute(classInitializeResult.StandardOut!, classInitializeResult.StandardError!, classInitializeResult.DebugTrace!, classInitializeResult.TestContextMessages!);
+                        result = testMethodRunner.Execute(classInitializeResult.StandardOut!, classInitializeResult.StandardError!, classInitializeResult.DebugTrace!, classInitializeResult.TestContextMessages!).ToUnitTestResults();
```

Comment:
@Evangelink My idea here is now that Execute returns TestResult, we can make the RetryAttribute work with TestResult as well, which is in TestFramework.

Basically, internally in Execute we were already dealing with TestResult but we converted it to UnitTestResult. I'm making it such that the conversion happens outside the Execute.

Does that make sense?